### PR TITLE
feature(ui-ux): token detail - reenable for DFI #2395 (#2758)

### PR DESCRIPTION
**Pull request by thedoublejay**
* token detail - reenable for DFI #2395

* token detail - reenable for DFI #2395

* token detail - reenable for DFI #2395

#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

**Comment by thedoublejay** [comment]
@rox1338 I have moved your PR here so we can run the tests. Some additional changes:
1. Can you change the onNavigateSwapDFI to use same method as onNavigateSwap? Maybe make the params optional so we don't recreate the same method. The usage is the same also.

Once #1 is done, you don't need to setIsFromToken anymore since it's passed as a parameter.

You can create the PR to this branch

**Comment by rox1338** [comment]
https://github.com/DeFiCh/wallet/pull/2772

**Comment by thedoublejay** [comment]
@rox1338 there's a type error on your last commit

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx [hunk 1]
```diff
@@ -105,12 +106,13 @@ export function TokenDetailScreen ({ route, navigation }: Props): JSX.Element {
     })
   }
 
-  const onNavigateSwap = ({ pair }: { pair: PoolPairData }): void => {
+  const onNavigateSwap = ({ pair, fromToken }: { pair: PoolPairData, fromToken: WalletToken }): void => {
```
Comment (rox1338) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx:109]:
```suggestion
  const onNavigateSwap = ({ pair, fromToken }: { pair?: PoolPairData, fromToken?: WalletToken }): void => {
```

**Review by rox1338 — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx (hunk 1 above)
Comment (rox1338) [reply mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx:109]: @thedoublejay that should fix it
I think you can commit it using github

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx [hunk 2]
```diff
@@ -126,6 +128,17 @@ export function TokenDetailScreen ({ route, navigation }: Props): JSX.Element {
...
+    navigation.navigate(translate('BottomTabNavigator', 'Balances'), {
+      screen: 'CompositeSwap',
+      initial: false,
+      params: {
+        fromToken: DFIUnified
+      },
+      merge: true
+    })
+  }
+
```
Comment (rox1338) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx:141]:
```suggestion
```

Comment (rox1338) [reply mobile-app/app/screens/AppNavigator/screens/Balances/screens/TokenDetailScreen.tsx:141]: For some reason this didn't get deleted. Must be some merge problem or sth.

**Comment by thedoublejay** [comment]
@rox1338 Some comments:

1. Token Detail - DeFiChain token link is wrong - when opening DeFiScan
2, Portfolio - DFI Border card is not rounded anymore 

When creating PR, please ensure that all changes will not cause a regression issue and it's well tested. Thanks!

**Comment by rox1338** [comment]
@thedoublejay I always try to test my changes. Sometimes I probably don't have the eye for some details as I am not as experienced with the codebase as you are.
@1: Didn't try that link because I didn't expect a problem here.
@2: I didn't see that because I used the bright theme.
@3: I tried to change the touch area to be the same as with the other cards. I think that is what you meant?

I tried to fix all the things and created a PR #2788 

Regarding the convert-button: Do we want to have both options "Convert to UTXO" and "Convert to Token" in the DFI token details? Or just one button called "Convert"? Or the way it is now: 1 button "Convert to UTXO"?
image

**Comment by thedoublejay** [comment]
> @thedoublejay I always try to test my changes. Sometimes I probably don't have the eye for some details as I am not as experienced with the codebase as you are. @1: Didn't try that link because I didn't expect a problem here. @2: I didn't see that because I used the bright theme. @3: I tried to change the touch area to be the same as with the other cards. I think that is what you meant?
> 
> I tried to fix all the things and created a PR #2788
> 
> Regarding the convert-button: Do we want to have both options "Convert to UTXO" and "Convert to Token" in the DFI token details? Or just one button called "Convert"? Or the way it is now: 1 button "Convert to UTXO"? image

@rox1338 Thanks for the prompt reply and changes. Don't get me wrong, I highly appreciate your contribution. Just wanted to align that testing is important, regardless of being experienced with codebase or not. It's ok we can keep it as Convert to UTXO first. Thanks again! Will test now

**Pull request merged**